flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/SumAggFunction.scala

```diff
@@ -49,13 +50,25 @@ abstract class SumAggFunction[T: Numeric] extends AggregateFunction[T] {
       val v = value.asInstanceOf[T]
       val a = accumulator.asInstanceOf[SumAccumulator[T]]
       a.f0 = numeric.plus(v, a.f0)
-      a.f1 = true
+      a.f1 += 1
+    }
+  }
+
+  override def retract(accumulator: Accumulator, value: Any): Unit = {
+    if (value != null) {
+      val v = value.asInstanceOf[T]
+      val a = accumulator.asInstanceOf[SumAccumulator[T]]
+      a.f0 = numeric.plus(v, a.f0)
+      a.f1 -= 1
+      if (a.f1 < 0) {
```

Comment:
Do we want to check for these errors? It adds overhead and is the responsibility of the runtime to call the functions correctly. I'd rather add tests with custom aggregation functions to verify the behavior instead of adding overhead to the most widely used aggregation functions. We didn't add these checks to `CountAggregate` and `AverageAggregate`.